pyiceberg/table/__init__.py

```diff
@@ -469,7 +474,9 @@ class SetLocationUpdate(TableUpdate):
 
 class SetPropertiesUpdate(TableUpdate):
     action: TableUpdateAction = TableUpdateAction.set_properties
-    updates: Dict[str, str]
+    updates: Properties
+    # validators
+    transform_properties_dict_value_to_str = field_validator('updates', mode='before')(transform_dict_value_to_str)
```

Comment:
```suggestion
    @field_validator('updates', mode='before')
    def transform_properties_dict_value_to_str(cls, properties: Properties) -> Dict[str, str]:
        return transform_dict_value_to_str(properties)
```
Shall we reformat the validator to use decorator? This aligns with the way we define model validators in `metadata.py` and also the example in [Pydantic doc](https://docs.pydantic.dev/latest/concepts/validators/#field-validators). WDYT?

Reply:
looks much cleaner, thank you!